shared-modules-auto-pr/entrypoint.py

```diff
@@ -0,0 +1,243 @@
+        subject, pr_message, base, head, maintainer_can_modify=True,
+    )
+    log.info("Opened pull request %s", pr.html_url)
+
+
+def bump_repo(repo: pygit2.Repository):
+    """
+        Update the repository submodules
+        commit the changes and create a merge request
+    """
+    submodule = repo.lookup_submodule(SHARED_MODULES)
+    shared_modules = submodule.open()
+    message = "Bump shared-modules"
+
+    subprocess.check_call(['git', 'pull', 'origin', 'master'],
+                          cwd=shared_modules.path)
+
+    merge_branch = 'flathubbot/shared-modues'
+
+    with indir(get_repo_dir(repo)):
+        subprocess.check_call(
+            ['git', 'config', '--global', 'user.name', '"flathubbot"'])
+        subprocess.check_call(['git', 'config', '--global', 'user.email',
+                               '"sysadmin@flathub.org"'])
+        subprocess.check_call(['git', 'checkout', 'HEAD@{0}'])
+        subprocess.check_call(['git', 'add', SHARED_MODULES])
+        subprocess.check_call(["git", "commit", "-am", message])
+
+        subprocess.check_call(['git', 'checkout', '-b', merge_branch])
+        open_pr(message, message, merge_branch)
+
+
+def detect_manifest(app_id: str, repo_path: pathlib.Path):
+    """
+        Detect and return the manifest path
+    """
+    exts = ["yml", "yaml", "json"]
+    for ext in exts:
+        manifest = repo_path.joinpath(f"{app_id}.{ext}")
+        if manifest.exists():
+            return manifest
+
+
+def get_yaml_shared_modules(modules: list) -> [str]:
+    shared_modules = []
+    for module in modules:
+        if isinstance(module, str) and module.startswith(SHARED_MODULES):
+            shared_modules.append(module)
+        elif isinstance(module, list):
+            shared_modules.extend(get_yaml_shared_modules(module))
+    return shared_modules
+
+
+def get_json_shared_modules(root: Json.Node) -> [str]:
+    shared_modules = []
+    elements = root.get_object().get_array_member('modules').get_elements()
+    for elem in elements:
+        module = elem.get_string()
+        if module and module.startswith(SHARED_MODULES):
+            shared_modules.append(module)
+    return shared_modules
+
+
+def parse_manifest(manifest_path: pathlib.Path) -> [str]:
+    ext = manifest_path.suffix
+    with open(manifest_path, 'r') as handle:
+        data = handle.read()
+
+    shared_modules = []
+
+    if ext in [".json"]:
+        parser = Json.Parser.new()
+        parser.load_from_data(data, len(data))
+        manifest_root = parser.get_root()
+        shared_modules = get_json_shared_modules(manifest_root)
+    elif ext in [".yml", ".yaml"]:
+        manifest = yaml.safe_load(data)
+        shared_modules = get_yaml_shared_modules(manifest['modules'])
+
+    return shared_modules
+
+
+if __name__ == "__main__":
+
+    github_token = os.environ['GITHUB_TOKEN']
+    g = Github(github_token, per_page=200)
+
+    org = g.get_organization('flathub')
+
+    for repo in org.get_repos():
```

Comment:
It may be better idea to just clone everything and then sort out which repositories have shared-modules configured, like external-data-checker action does.

Reply:
It should be doing that